File: 162. Find Peak Element/162.go
// 162. Find Peak Element solution in GoLang
// https://leetcode.com/problems/find-peak-element/description/

package leetcode

func findPeakElement(nums []int) int {
    low := 0
    high := len(nums) - 1

    for low < high {
        mid := low + ((high - low) / 2)

        if nums[mid] < nums[mid + 1] {
            low = mid + 1
        } else {
            high = mid
        }
    }

    return low
}

File: 162. Find Peak Element/162.js
// 162. Find Peak Element solution in JavaScript
// https://leetcode.com/problems/find-peak-element/description/

/**
 * @param {number[]} nums
 * @return {number}
 */
var findPeakElement = function(nums) {
    let low = 0;
    let high = nums.length - 1;

    while (low < high) {
        let mid = low + Math.floor((high - low) / 2);

        if (nums[mid] > nums[mid + 1]) {
            high = mid;
        } else {
            low = mid + 1;
        }
    }

    return low;
};